go/billing/api.py

```diff
@@ -493,3 +510,32 @@ def getChild(self, name, request):
     def render_GET(self, request):
         request.setResponseCode(200)  # OK
         return ''
+
+    @defer.inlineCallbacks
+    def health_check(self, request):
+        """
+        We want our health check to be comprehensive, so we implement it here.
+        """
+        try:
+            yield self._connection_pool.runQuery("SELECT 1")
+        except (ConnectionDead, psycopg2.InterfaceError):
+            request.setResponseCode(503)
+            request.write("Database connection unavailable\n")
+        else:
+            request.setResponseCode(200)
+            request.write("OK\n")
+        request.finish()
+
+
+def billing_api_resource():
+    """
+    Create and return a go.billing.api.Root resource for use with twistd.
+    """
+    connection_string = app_settings.get_connection_string()
+    connection_pool = DictRowConnectionPool(
+        None, connection_string, min=app_settings.API_MIN_CONNECTIONS)
+    resource = Root(connection_pool)
+    # Tests need to know when we're connected, so stash the deferred on the
+    # resource for them to look at.
+    resource._connection_pool_started = connection_pool.start()
```

Comment:
It feels like the reason we're having to stash ``_connection_pool_started`` is because this method is really doing two completely different things (creating and starting a connection pool and creating an api Root resource). Perhaps it should only be doing one thing and we should have a separate method for creating the connection pool?